### PR TITLE
Only count data that is in memory for xarray sizeof

### PR DESCRIPTION
**Pull request by fjetter**
- [x] Closes https://github.com/dask/dask/issues/11204
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

cc @dcherian

### CODE REVIEW

File: dask/sizeof.py [hunk 1]
```diff
@@ -267,10 +267,18 @@ def register_xarray():
...
     @sizeof.register(xr.DataArray)
-    @sizeof.register(xr.Dataset)
+    @sizeof.register(xr.Variable)
     def xarray_sizeof_da(obj):
-        return obj.nbytes
+        return sys.getsizeof(obj) + sizeof(obj.data)
+
+    @sizeof.register(xr.Dataset)
+    def xarray_sizeof_ds(obj):
+        return sys.getsizeof(obj) + sizeof(obj.variables)
```
Comment (fjetter) [inline dask/sizeof.py:281]: I guess there is stuff in a Dataset that I am not tracking here. @dcherian I would appreciate a review. If we're off by a couple percent that's not too bad but we'd like to capture the bulk of the data if possible

**Review by jrbourbeau — COMMENTED** [review]
@dcherian do you have bandwidth to take a look at this one? For reference, we're releasing on Friday, so this could get pushed out to users soon

**Comment by fjetter** [comment]
> we're releasing on Friday, so this could get pushed out to users soon

I'm pretty certain this is better behavior than what we've had before. I'm not sure if it quite right. Including this in the release makes sense even if we don't get any response

File: dask/sizeof.py [hunk 2]
```diff
@@ -267,10 +267,18 @@ def register_xarray():
     XARRAY_VERSION = Version(xr.__version__)
     XARRAY_GE_2024_02 = XARRAY_VERSION >= Version("2024.02.0")
 
+    @sizeof.register(xr.core.utils.Frozen)
+    def xarray_sizeof_frozen(obj):
+        return sys.getsizeof(obj) + sizeof(obj.mapping)
+
     @sizeof.register(xr.DataArray)
-    @sizeof.register(xr.Dataset)
+    @sizeof.register(xr.Variable)
```
Comment (dcherian) [inline dask/sizeof.py:275]:
```suggestion
```

**Comment by dcherian** [comment]
Can you merge in #11216 here? it should make it better but not perfect.

**Comment by jrbourbeau** [comment]
@dcherian it looks like there are a couple related tests failing over there 

```
FAILED dask/tests/test_sizeof.py::test_xarray - AttributeError: 'DataArray' object has no attribute '_attrs'
FAILED dask/tests/test_sizeof.py::test_xarray_not_in_memory - assert 2758 < (1064 + 1064)
 +  where 2758 = sizeof(<xarray.Dataset> Size: 2kB\nDimensions:  (coord: 133)\nCoordinates:\n  * coord    (coord) float64 1kB -66.0 -65.0 -64.0 -63.0 ... 63.0 64.0 65.0 66.0\nData variables:\n    foo      (coord) float64 1kB dask.array<chunksize=(133,), meta=np.ndarray>)
 +  and   1064 = sizeof(array([0.3528154 , 0.60319957, 0.29865345, 0.70480754, 0.59733608,\n       0.72447864, 0.74871366, 0.726466  , 0.87247927, 0.96198241,\n       0.46793289, 0.27200241, 0.50235041, 0.0851269 , 0.33977135,\n       0.31437367, 0.59382297, 0.32812972, 0.67662178, 0.48966544,\n       0.89576915, 0.12326049, 0.01017166, 0.47198195, 0.26124414,\n       0.42421292, 0.98335093, 0.93210694, 0.36366348, 0.68818795,\n       0.33410805, 0.41169089, 0.05835039, 0.12875558, 0.76681677,\n       0.97345747, 0.77020028, 0.91660321, 0.31011189, 0.37412729,\n       0.05716177, 0.05745778, 0.95292922, 0.21567564, 0.22655973,\n       0.26843991, 0.3100695 , 0.49382991, 0.40322648, 0.05130205,\n       0.29161665, 0.47468751, 0.83622982, 0.97634814, 0.12904219,\n       0.30696832, 0.04246488, 0.17437984, 0.71956203, 0.82204763,\n       0.07511585, 0.73464722, 0.42062425, 0.39828281, 0.9865718 ,\n       0.36654268, 0.10048549, 0.18726196, 0.62680319, 0.73434468,\n       0.6142511 , 0.75606326, 0.50213386, 0.63405037, 0.0026173 ,\n       0.02379692, 0.33477352, 0.26737613, 0.41262592, 0.5197189 ,\n       0.18986641, 0.87923963, 0.43984938, 0.49119927, 0.08736517,\n       0.24706035, 0.79339378, 0.48710167, 0.25729745, 0.08600394,\n       0.8311682 , 0.81402591, 0.40872157, 0.17638705, 0.45319758,\n       0.28685454, 0.50885428, 0.41931051, 0.55034345, 0.64501709,\n       0.25265896, 0.43511811, 0.54129786, 0.15615727, 0.4429807 ,\n       0.3250982 , 0.75749494, 0.36707466, 0.19844491, 0.21857109,\n       0.39221173, 0.52428935, 0.51447594, 0.4442796 , 0.70454905,\n       0.4102104 , 0.13106824, 0.14564531, 0.67406767, 0.63789323,\n       0.69860021, 0.19727814, 0.42596478, 0.19447004, 0.30838348,\n       0.23201388, 0.03032531, 0.48191989, 0.47925421, 0.38746641,\n       0.09747502, 0.46851099, 0.57303647]))
 +  and   1064 = sizeof(array([-66., -65., -64., -63., -62., -61., -60., -59., -58., -57., -56.,\n       -55., -54., -53., -52., -51., -50., -49., -48., -47., -46., -45.,\n       -44., -43., -42., -41., -40., -39., -38., -37., -36., -35., -34.,\n       -33., -32., -31., -30., -29., -28., -27., -26., -25., -24., -23.,\n       -22., -21., -20., -19., -18., -17., -16., -15., -14., -13., -12.,\n       -11., -10.,  -9.,  -8.,  -7.,  -6.,  -5.,  -4.,  -3.,  -2.,  -1.,\n         0.,   1.,   2.,   3.,   4.,   5.,   6.,   7.,   8.,   9.,  10.,\n        11.,  12.,  13.,  14.,  15.,  16.,  17.,  18.,  19.,  20.,  21.,\n        22.,  23.,  24.,  25.,  26.,  27.,  28.,  29.,  30.,  31.,  32.,\n        33.,  34.,  35.,  36.,  37.,  38.,  39.,  40.,  41.,  42.,  43.,\n        44.,  45.,  46.,  47.,  48.,  49.,  50.,  51.,  52.,  53.,  54.,\n        55.,  56.,  57.,  58.,  59.,  60.,  61.,  62.,  63.,  64.,  65.,\n        66.]))
```

**Comment by jrbourbeau** [comment]
I'm going to go ahead and merge this PR in as is so it goes into the release today. The changes here seem like a clear improvement over the current behavior on `main` and we can definitely follow up with updates to continue to improve the situation.

**Pull request merged**